core/services/relay/evm/codec_test.go

```diff
@@ -0,0 +1,189 @@
+package evm_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	. "github.com/smartcontractkit/chainlink-common/pkg/types/interfacetests" //nolint common practice to import test mods with .
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/testfiles"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+func TestCodec(t *testing.T) {
+	RunCodecInterfaceTests(t, &codecInterfaceTester{})
+	anyN := 10
+	t.Run("GetMaxEncodingSize delegates to GetMaxSize", func(t *testing.T) {
+		codec := getCodec(t)
+
+		actual, err := codec.GetMaxEncodingSize(context.Background(), anyN, sizeItemType)
```

Comment:
We should be using `testutils.Context(t)` instead of `context.Background()` for tests to avoid leaked goroutines

Reply:
Oh nevermind, I guess in this case it doesn't matter since the context is unused:
```
func (e *encoder) GetMaxEncodingSize(ctx context.Context, n int, itemType string) (int, error) {
    return e.Definitions[itemType].GetMaxSize(n)
}
```

I'm still trying to wrap my head around the whole gRPC paradigm. So, the contexts in the signatures for methods like this are mostly for the client side, so it can abort while waiting on I/O... and then the contexts in the signatures for the methods on the server side are mostly required in order to use the same interface on both sides? (But might also be used, if there is some I/O operation like reading from the db?

Reply:
I'll use it for consistency.  I only learned about it recently and didn't think about it again when I was picking files from other commits :).